dashboard/app/models/course.rb

```diff
@@ -319,8 +317,13 @@ def summarize_short
   # sharing the family_name of this course, including this one.
   def summarize_versions(user = nil)
     return [] unless family_name
-    versions = Course.
-      where("properties -> '$.family_name' = ?", family_name).
+
+    # Include visible courses, plus self if not already included
+    courses = Course.valid_courses(user: user).clone
```

Comment:
interesting. it is hard to stop brainstorming reasons for the other bug, where CSP/CSD 20-21 are appearing the EditSectionDialog / AssignmentSelector. I suppose it is possible that we are modifying the cached value there. Could this be the cause of the other bug, on this line, because concat modifies the value in the cache? https://github.com/code-dot-org/code-dot-org/blob/fe1396dc10c98965523cce66b883b706d79bb63a/dashboard/app/models/course.rb#L242

Reply:
Oh wow, that actually could be it!